python/plugins/processing/gui/AlgorithmDialog.py

```diff
@@ -121,6 +121,18 @@ def setParamValues(self):
 
         return True
 
+    def evaluateExpression(self, text):
+        context = QgsExpressionContext()
+        context.appendScope(QgsExpressionContextUtils.projectScope())
```

Comment:
you should first add the global scope:
context.appendScope(QgsExpressionContextUtils.globalScope())